pytorch_lightning/utilities/auto_restart.py

```diff
@@ -571,3 +572,15 @@ def reload_dataloader_state_dict(dataloader: DataLoader, state_dict: Dict[str, A
 
     else:
         raise MisconfigurationException("This shouldn't happen. Please, open an issue on PyTorch Lightning Github.")
+
+
+def is_obj_stateful(obj: Any) -> bool:
```

Comment:
Python 3.8 supports Protocols which you might find helpful here: https://www.python.org/dev/peps/pep-0544/

Reply:
I checked it out and it seemed an overkill.

Reply:
I think that's a great suggestion which is also Pythonic and is designed exactly for the check you want to implement.

We are not limited by 3.8 because it's available in `typing_extensions` which we already install

Reply:
Pushed d04596d